earth_enterprise/src/fusion/autoingest/StorageManager.h

```diff
@@ -355,5 +360,11 @@ bool StorageManager<AssetType>::SaveDirtyToDotNew(
   return true;
 }
 
+template<class AssetType>
+bool StorageManager<AssetType>::DetermineIfPrune() {
+  std::lock_guard<std::recursive_mutex> lock(storageMutex);
+  return ((float)DirtySize() < ((float)CacheSize() * (1 - maxPurgePercent)));
```

Comment:
A few small comments:

I'd recommend changing this to `<=`. That way, if we set maxPurgePercent to 0, this function will always return true (which allows us to get the behavior we get without these changes).

I'm not sure that casting to `float` is necessary. Sure, integer math won't be as exact, but it's probably close enough for this purpose.

The lock is not strictly necessary in this function. Everywhere it is called the code will have already acquired the lock.

Reply:
In a previous comment you mentioned that we do not want to prune the cache if the dirty map and cache size are the same. Changing this condition to <= would cause the prune condition to be met if maxPurgePercent is also set to 0.

Reply:
The reasoning behind this suggestion is that we're adding a feature that may make Fusion faster, but it may not. So, I think there should be a way to turn it off completely. Users will have the option to set maxPurgePercent to 0 and get the old behavior, or they have the option to set it to a different number and possibly get better performance (or possibly get worse performance if it doesn't work as well as I think it will). So, overall, I think it will work fastest if we don't purge when the dirty map and the cache are the same size, but I want to give users that option in case I'm wrong.

Reply:
I'm in the process of changing this to **CacheSize() * maxPurgePercent**. With that in mind, **PurgePercent** now has the opposite affect. If we keep the condition `<` as opposed to `<=`and set the **PurgePercent** to 1, the cache will be pruned as long as the sizes are not the same.

Reply:
I hadn't seen your comment, I am now changing the condition to `<=`.